app.rb

```diff
@@ -0,0 +1,130 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './rental'
+require_relative './book'
+require_relative './classroom'
+
+class App
+  attr_accessor :people, :books, :rentals, :classroom
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def book_list
+    puts 'No book!' if @books.empty?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def person_list
+    puts 'You have not registered any person' if @people.empty?
+    @people.each { |person| puts "[#{person.class}], Name: #{person.name}, Age: #{person.age}, Id: #{person.id}" }
+  end
+
+  def create_person
+    print 'enter (1) to create student or (2) to create teacher? [Input the number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      'You have entered an invalid option'
+    end
+    puts 'You have successfully registered person!'
+  end
+
+  def create_teacher
+    print "teacher's specialization: "
+    specialization = gets.chomp
+    print "teacher's age: "
+    age = gets.chomp
+    print "teacher's name: "
+    name = gets.chomp
+    teacher = Teacher.new(specialization, age, name, parent_permission: true)
+    puts 'You have successfully registered a teacher'
+    @people.push(teacher)
+  end
+
+  def create_student
+    print "students's age: "
+    age = gets.chomp
+
+    print "students's name: "
+    name = gets.chomp
+
+    print 'has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.upcase
+
+    case parent_permission
+    when 'N'
+      student = Student.new(nil, age, name, parent_permission: false)
+      @people.push(student)
+    when 'Y'
+      student = Student.new(nil, age, name, parent_permission: true)
+      @people.push(student)
+    else
+      'You have entered an invalid option'
+    end
+
+    puts 'you have successfully registered a student!'
+  end
+
+  def create_book
+    print "book's title: "
+    title = gets.chomp
+
+    print "book's author: "
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books.push(book)
+
+    puts 'You have successfuly added a book!'
+  end
+
+  def create_rental
+    if @books.size.zero?
+      puts 'No book available'
+
+    elsif @people.size.zero?
+      puts 'No person available'
+
+    else
+      puts 'Select a book by number'
+      @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+      selected_book = gets.chomp.to_i
+
+      puts 'Select a person by number'
+      @people.each_with_index do |person, index|
+        puts "#{index}) Name: #{person.name} Age: #{person.age} Id: #{person.id}"
+      end
+
+      selected_person = gets.chomp.to_i
+
+      puts 'Enter date [YYYY-MM-DD]'
+      selected_date = gets.chomp.to_s
+
+      rental_item = Rental.new(selected_date, @people[selected_person], @books[selected_book])
```

Comment:
- [ ] Your app is throwing an error in an attempt to create a rental. An error is thrown just after entering details to create a rental. Kindly resolve the same.

**Creating a rental error:**
![image](https://user-images.githubusercontent.com/12745474/193359155-8a21283e-def0-4945-a52b-a8b61e36d577.png)

Reply:
Thanks a lot, @rloterh  for spotting this issue  I will do well to fix the issue.